### PR TITLE
feat: Convert grid view to Rail and Rail to Rail v3

### PR DESCRIPTION
**Pull request by rajsam003**
The type of this PR is: Enhancement

This PR resolves [CX-1818]

### Description

This PR resolves the following
- Convert home feed artwork rail into rail v3 
- Convert home artworks grid view to rail v3

|Before|After|
|---|---|
|image|image|
|image|image|

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [x] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Convert home feed artwork rail into rail v3 - sam
- Convert home feed artworks grid view to rail v3 - sam

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[CX-1818]: https://artsyproduct.atlassian.net/browse/CX-1818?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 27d63c442e2fdf389b849a260c57ab4a195910ce
</p>

File: src/lib/Scenes/Home/Components/ArtworkRail.tsx [hunk 1]
```diff
@@ -93,25 +92,12 @@ const ArtworkRail: React.FC<ArtworkRailProps & RailScrollProps> = ({ title, rail
...
-            trackTap={(artworkSlug, index) => {
-              const tapEvent = HomeAnalytics.artworkThumbnailTapEventFromKey(rail.key, artworkSlug, index)
-              if (tapEvent) {
-                tracking.trackEvent(tapEvent)
-              }
-            }}
-          />
-        </Box>
-      )}
+      <SmallTileRailContainer
```
Comment (rajsam003) [inline src/lib/Scenes/Home/Components/ArtworkRail.tsx:95]: I can change the name of this component to a more generic one like `ArtworkTileRail` but it already exists. 
I tried to use it but then it required a big change and a lot of files are getting affected.

File: src/lib/Scenes/Home/Home.tsx [hunk 2]
```diff
@@ -520,6 +520,24 @@ const BelowTheFoldPlaceholder: React.FC = () => {
...
+    },
+    large: {
+      width: 295,
+      maxHeight: 320,
+    },
+  }
+  const smallImagePlaceHolderWidth = IMAGE_PLACE_HOLDER_SIZES.small.width
+  const largeImagePlaceHolderWidth = IMAGE_PLACE_HOLDER_SIZES.large.width
+
+  const imagePlaceHolderAspectRatio = 0.7
```
Comment (rajsam003) [inline src/lib/Scenes/Home/Home.tsx:536]: Not sure about this. We need to ask Jonathan about this command aspect ratio to use for placeholder box

File: src/lib/Scenes/Home/Home.tsx [hunk 3]
```diff
@@ -520,6 +520,24 @@ const BelowTheFoldPlaceholder: React.FC = () => {
 const HomePlaceholder: React.FC<{}> = () => {
   const enableViewingRooms = useFeatureFlag("AREnableViewingRooms")
 
+  const IMAGE_PLACE_HOLDER_SIZES = {
```
Comment (olerichter00) [inline src/lib/Scenes/Home/Home.tsx:523]:
```suggestion
  const IMAGE_PLACEHOLDER_SIZES = {
```

File: src/lib/Scenes/Home/Home.tsx (hunk 2 above)
Comment (olerichter00) [inline src/lib/Scenes/Home/Home.tsx:536]:
```suggestion
  const imagePlaceholderAspectRatio = 0.7
```

File: src/lib/Scenes/Home/Home.tsx [hunk 4]
```diff
@@ -520,6 +520,24 @@ const BelowTheFoldPlaceholder: React.FC = () => {
...
+      width: 155,
+      maxHeight: 230,
+    },
+    large: {
+      width: 295,
+      maxHeight: 320,
+    },
+  }
+  const smallImagePlaceHolderWidth = IMAGE_PLACE_HOLDER_SIZES.small.width
+  const largeImagePlaceHolderWidth = IMAGE_PLACE_HOLDER_SIZES.large.width
```
Comment (olerichter00) [inline src/lib/Scenes/Home/Home.tsx:534]:
```suggestion
  const smallImagePlaceholderWidth = IMAGE_PLACEHOLDER_SIZES.small.width
  const largeImagePlaceholderWidth = IMAGE_PLACEHOLDER_SIZES.large.width
```

**Review by olerichter00 — CHANGES_REQUESTED** [review]
I love this feature and the code looks good to me 🌟 

I added a few comments about the naming and I get layout shifts when I scroll on the rail (iOS simulator iPhone 12 mini). Please also make sure this looks good on Android and on tablets. I guess it will look as expected but we should check (and add screenshots to the PR).

File: src/lib/Components/ArtworkTileRail/ArtworkTileRailCard.tsx [hunk 5]
```diff
@@ -8,9 +8,18 @@ import styled from "styled-components/native"
...
-  small: 120,
-  medium: 160,
-  large: 240,
+  small: {
+    width: 155,
+    maxHeight: 230,
+  },
+  medium: {
+    width: 160,
+    maxHeight: 275, // Calcuated middle number of maxHeight small and large.
```
Comment (pvinis) [inline src/lib/Components/ArtworkTileRail/ArtworkTileRailCard.tsx:17]:
```suggestion
    maxHeight: 275, // Calculated middle number of maxHeight small and large.
```

**Review by olerichter00 — CHANGES_REQUESTED** [review]
I can still see the same layout shifts. This is probably due to the rail loading more artworks when scrolling which then increases the height. I guess, we either need to set the height of the images to a fixed value (setting `height` instead of `maxHeight`) or disable load more on scroll for all artwork rails.

File: src/lib/Scenes/Home/Home.tsx [hunk 6]
```diff
@@ -246,7 +246,6 @@ const Home = (props: Props) => {
                     title={item.title}
                     rail={item.data || null}
                     scrollRef={scrollRefs.current[index]}
-                    mb={MODULE_SEPARATOR_HEIGHT}
```
Comment (rajsam003) [inline src/lib/Scenes/Home/Home.tsx:249]: Removed it as it gives extra unwanted space after adding `height={90}` for the text container of rails as per design,

Comment (olerichter00) [reply src/lib/Scenes/Home/Home.tsx:249]: Giving the text below the images in the artwork item a fixed height of 90px would result in different spaces between rails dependent on the content of the text, right? I'd argue it is better to keep this bottom-margin and removing the fixed height of 90px.

**Review by olerichter00 — CHANGES_REQUESTED** [review]
Code looks good 👍 
I still get a few layout shifts just after the rail renders for the first time.

**Comment by rajsam003** [comment]
> I still get a few layout shifts just after the rail renders for the first time (iOS simulator).

@olerichter00 this is not happening on my end. It happens only before the initial render which I think is expected.

**Comment by olerichter00** [comment]
> > I still get a few layout shifts just after the rail renders for the first time (iOS simulator).
> 
> @olerichter00 this is not happening on my end. It happens only before the initial render which I think is expected.

I recorded it (iOS 15.2 iPhone 12 mini Simulator):

https://user-images.githubusercontent.com/4691889/148581404-b6e56fb0-ad43-4537-abe2-440b4fde7d52.mp4

**Comment by rajsam003** [comment]
> > > I still get a few layout shifts just after the rail renders for the first time (iOS simulator).
> > 
> > 
> > @olerichter00 this is not happening on my end. It happens only before the initial render which I think is expected.
> 
> I recorded it (iOS 15.2 iPhone 12 mini simulator):
> 
>  Simulator.Screen.Recording.-.iPhone.12.mini.-.2022-01-07.at.18.16.21.mp4

Yes, this is what I meant. If this is not expected, I will have to figure out how to avoid this.

**Comment by olerichter00** [comment]
> Yes, this is what I meant. If this is not expected, I will have to figure out how to avoid this.

I'd say we should fix the issue before shipping these new rails.

**Comment by olerichter00** [comment]
Closing this PR, the new artwork rail was implemented in https://github.com/artsy/eigen/pull/6045.

**Pull request closed without merging**